src/util/RatelimitBucket.hpp

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <QList>
+#include <QObject>
+#include <QString>
+
+namespace chatterino {
+
+class RatelimitBucket : public QObject
+{
+public:
+    RatelimitBucket(int budget, int cooldown,
+                    std::function<void(QString)> callback, QObject *parent);
+
+    void send(QString message);
+
+private:
+    /**
+     * @brief budget_ denotes the amount of calls that can be handled before we need to wait for the cooldown
+     **/
+    int budget_;
+
+    /**
+     * @brief This is the amount of time in milliseconds it takes for one used up budget to be put back into the bucket for use elsewhere
+     **/
+    int cooldown_;
```

Comment:
```suggestion
    const int cooldown_;
```
It makes no sense for the cooldown to change.